### PR TITLE
[Fix/#66]: Profile 관련 타입 재설정 및 기타 수정

### PR DESCRIPTION
**Pull request by SebellKo**
## 📌 작업 내용

- [x] Profile 관련 타입 재설정
- [x] 사용자 타입 판별 커스텀훅 구현

멘토님께서 조언해주신 Profile 관련 타입과 유저 타입 판별을 Boolean으로 적용할것 등 리팩토링을 진행했습니다.

<br/>

## 📌 구현 결과 (선택)

<br/>

## 📌 기타 사항

<br/>

### CODE REVIEW

**Comment by JIMIN1020** [comment]
<h2>Coverage Report</h2>
<table> <thead> <tr> <th align="center">Status</th> <th align="left">Category</th> <th align="right">Percentage</th> <th align="right">Covered / Total</th> </tr> </thead> <tbody> <tr> <td align="center">🔵</td> <td align="left">Lines</td> <td align="right">0%</td> <td align="right">0 / 4492</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Statements</td> <td align="right">0%</td> <td align="right">0 / 4492</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Functions</td> <td align="right">3.12%</td> <td align="right">5 / 160</td> </tr> <tr> <td align="center">🔵</td> <td align="left">Branches</td> <td align="right">3.12%</td> <td align="right">5 / 160</td> </tr> </tbody> </table><details><summary>File Coverage</summary><table> <thead> <tr> <th align="left">File</th> <th align="right">Stmts</th> <th align="right">% Branch</th> <th align="right">% Funcs</th> <th align="right">% Lines</th> <th align="left">Uncovered Lines</th> </tr> </thead> <tbody> <tr> <td align="left" colspan="6"><b>Changed Files</b></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx">src/features/Profile/components/EditProfile/CategoryButtons.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L1-L5" class="text-red">1-5</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L15-L16" class="text-red">15-16</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L18-L21" class="text-red">18-21</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L23-L36" class="text-red">23-36</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L38-L39" class="text-red">38-39</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L41-L43" class="text-red">41-43</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L45" class="text-red">45</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L47" class="text-red">47</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L53" class="text-red">53</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/CategoryButtons.tsx#L59" class="text-red">59</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx">src/features/Profile/components/EditProfile/EditProfileForm.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L1-L11" class="text-red">1-11</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L15-L25" class="text-red">15-25</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L32" class="text-red">32</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L34-L42" class="text-red">34-42</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L44-L59" class="text-red">44-59</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L61-L62" class="text-red">61-62</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L64-L66" class="text-red">64-66</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L68-L76" class="text-red">68-76</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L78-L99" class="text-red">78-99</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L101-L110" class="text-red">101-110</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L112-L119" class="text-red">112-119</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L121-L125" class="text-red">121-125</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L127" class="text-red">127</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L129" class="text-red">129</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L135-L136" class="text-red">135-136</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L142" class="text-red">142</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L146" class="text-red">146</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L154" class="text-red">154</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L160" class="text-red">160</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L166" class="text-red">166</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L171" class="text-red">171</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L176-L177" class="text-red">176-177</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/EditProfile/EditProfileForm.tsx#L179" class="text-red">179</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx">src/features/Profile/components/Mypage/MyPage.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx#L1-L7" class="text-red">1-7</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx#L13-L16" class="text-red">13-16</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx#L18" class="text-red">18</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx#L20" class="text-red">20</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx#L22-L31" class="text-red">22-31</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx#L33-L37" class="text-red">33-37</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx#L39" class="text-red">39</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/Mypage/MyPage.tsx#L41" class="text-red">41</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx">src/features/Profile/components/common/Profile.tsx</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L1-L3" class="text-red">1-3</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L10-L11" class="text-red">10-11</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L13-L27" class="text-red">13-27</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L29-L30" class="text-red">29-30</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L32" class="text-red">32</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L34" class="text-red">34</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L42" class="text-red">42</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L52" class="text-red">52</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L59-L61" class="text-red">59-61</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L64-L66" class="text-red">64-66</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/components/common/Profile.tsx#L69" class="text-red">69</a></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/features/Profile/model/profile.model.ts">src/features/Profile/model/profile.model.ts</a></td> <td align="right">100%</td> <td align="right">100%</td> <td align="right">100%</td> <td align="right">100%</td> <td align="left"></td> </tr> <tr> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/hooks/usertype/useUserType.ts">src/hooks/usertype/useUserType.ts</a></td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="right">0%</td> <td align="left"><a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/hooks/usertype/useUserType.ts#L1" class="text-red">1</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/hooks/usertype/useUserType.ts#L3-L7" class="text-red">3-7</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/hooks/usertype/useUserType.ts#L9-L10" class="text-red">9-10</a>, <a href="https://github.com/Licruit/Licruit-frontend/blob/a2ddf5e4c3774eecda8eaf92069606978f9cc8c2/src/hooks/usertype/useUserType.ts#L12" class="text-red">12</a></td> </tr> </tbody> </table></details>
<em>Generated in workflow <a href=https://github.com/Licruit/Licruit-frontend/actions/runs/10348947011>#92</a></em>

**Review by juhee067 — APPROVED** [review]
세종님 늘 고생많으십니다

**Review by JIMIN1020 — APPROVED** [review]
고생 많으셨습니다~~

File: src/features/Profile/components/EditProfile/EditProfileForm.tsx [hunk 1]
```diff
@@ -58,7 +59,7 @@ function EditProfileForm({ userProfile, image }: Props) {
   } = methods;
 
   const introduceValue = watch('introduce') || '';
-  const isCompany = sessionStorage.getItem(STORAGE_KEY.userType) === 'true';
+  const isCompany = checkIsCompany();
```
Comment (ksy90101) [inline src/features/Profile/components/EditProfile/EditProfileForm.tsx:62]: get이 더어울리는 이름일거 같아요.

File: src/features/Profile/components/Mypage/MyPage.tsx [hunk 2]
```diff
@@ -12,7 +12,8 @@ interface Props {
 
 function MyPage({ onClose }: Props) {
   const { data: userProfile, isError } = useProfileQuery();
-  const isCompany = sessionStorage.getItem(STORAGE_KEY.userType) === 'true';
+  const checkIsCompany = useUserType();
```
Comment (ksy90101) [inline src/features/Profile/components/Mypage/MyPage.tsx:15]: useUserType인데 checkIsCompany가 리턴값인게 이상하네요?
차라리 해당 부분을 객체로 리턴하는게 맞지 않나 생각이 드네요.

File: src/hooks/usertype/useUserType.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+import { STORAGE_KEY } from '@/constants/storage';
+
+const useUserType = () => {
```
Comment (ksy90101) [inline src/hooks/usertype/useUserType.ts:3]: 흠... 뭔가 코드가...
아튼 객체로 리턴해서 구조분해할당으로 빼는게 더 맞느넉 같아요.

File: src/features/Profile/components/Mypage/MyPage.tsx (hunk 2 above)
Comment (SebellKo) [reply src/features/Profile/components/Mypage/MyPage.tsx:15]: 확인했습니다 !

File: src/features/Profile/components/EditProfile/EditProfileForm.tsx (hunk 1 above)
Comment (SebellKo) [reply src/features/Profile/components/EditProfile/EditProfileForm.tsx:62]: get 확인입니다 !

File: src/hooks/usertype/useUserType.ts (hunk 3 above)
Comment (SebellKo) [reply src/hooks/usertype/useUserType.ts:3]: 넵 !

**Pull request merged**